docs/src/references/c8y-configuration-management.md

```diff
@@ -163,3 +278,167 @@ Note that:
+and all the connections to thin-edge are established by the child devices.
+* Thin-edge opens two ports for MQTT and HTTP over the local network.
+  These ports are controlled on the main device `tedge config`:
+  * `mqtt.external.bind_address`
+  * `mqtt.external.port`
+  * `http.external.port`
+* The child devices must know the main device IP address.
+  * This is the address set for `mqtt.external.bind_address` on the main device.
+  * For the very specific case, where the software for the child device run on the main device,
+    this connection address can be the `localhost`.
+* On start, the child-device specific software for configuration management,
+  must connect over MQTT to thin-edge device.
+  * It must use a session name that is unique on the thin-edge bus.
+    `"$CHILD_DEVICE_ID/configuration"` is the recommendation.
+  * It must subscribe to `tedge/$CHILD_DEVICE_ID/commands/req/#` to receive requests.
+  * It has to publish responses under `tedge/$CHILD_DEVICE_ID/commands/res/#`.
+* On demand, the child-device software has to send HTTP requests to thin-edge.
+  * These requests are REST requests (GET/PUT/DELETE) to exchange files.
+  * The urls are forged by thin-edge telling the child devices
+    where to get and put configuration files.
+* Currently, all these connections are done without TLS support.
+  * Hence, the main device must be configured so these MQTT and HTTP ports
+    are only open on a local trusted network.
+
+### The child device downloads configuration updates on notification
+
+When a configuration file for a child device
+is received from the cloud by the `c8y_configuration_plugin` daemon,
+the configuration plugin manages the transfer of this file to the child device
+and notifies the cloud on the progress of this configuration update operation.
+
+1. On reception of a configuration update for a child device named `$CHILD_DEVICE_ID`
+   of a file `{ path = $PATH, type = $TYPE }` defined in `$CHILD_DEVICE_ID/c8y-configuration-plugin.toml` 
+   the `c8y_configuration_plugin` downloads the new configuration content in `/tmp`,
+   and, on success, moves this content under the thin-edge file transfer HTTP root.
+   * The file is moved to `$TEDGE_HTTP_ROOT/$CHILD_DEVICE_ID/config_update/$TYPE`
+     making this file available under `http://$TEDGE_HTTP/tedge/$CHILD_DEVICE_ID/config_update/$TYPE`.
+   * Note that the `$TYPE` is by default the `$PATH` if not provided by the configuration.  
+1. Once the updated configuration is available over the local HTTP file transfer service,
+   the `c8y_configuration_plugin` notify the child device by publishing an MQTT message.
+   * The topic is `tedge/$CHILD_DEVICE_ID/commands/req/config_update`
+   * The payload is a JSON record with 3 fields
+     * `"url": "http://$TEDGE_HTTP/tedge/$CHILD_DEVICE_ID/config_update/$TYPE"`
+     * `"path": "$PATH"`
+     * `"type": "$TYPE"` (if no `type` has been specified, then this field is omitted)
```

Comment:
If no "$TYPE" is specified, then the "url" line cannot be defined as described above.

Should it say  $PATH  instead?

```suggestion
   * The payload is a JSON record with 3 fields
     * `"url": "http://$TEDGE_HTTP/tedge/$CHILD_DEVICE_ID/config_update/$PATH"`
     * `"path": "$PATH"`
     * `"type": "$TYPE"` (if no `type` has been specified, then this field is omitted)
```

Reply:
When the `type` is not given explicitly one have to use the `path` instead. Giving the user to option of a `type` that is not the `path` can help when the actual path contains characters that are not easy to handle in an HTTP url or an MQTT topic.

Reply:
I think we just need one note there, if $TYPE is not given, use $PATH in the `url`.